spec/requests/players_request_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "rails_helper"
+
+RSpec.describe "Players", type: :request do
+  let(:user) { FactoryBot.create(:user) }
+  let(:game) { FactoryBot.create(:game)}
+  let(:rank) { FactoryBot.create(:rank, game: game)}
+  let(:other_game) { FactoryBot.create(:game)}
+  let(:other_rank) { FactoryBot.create(:rank, game: other_game)}
+
+  before(:each) do
+    log_in_as user
+  end
+
+  describe "GET players#index" do
+    it "should get index" do
+      get players_path
+      expect(response).to have_http_status(200)
+    end
+  end
+
+  describe "POST player#create" do
+    it "create player with valid attributes" do
+      player_params = {
+        player: {
+          game_id: game.id,
+          rank_id: rank.id
+        }
+      }
+      expect {post "/players", params: player_params}
+        .to change{Player.count}.by(1)
+    end
+  end
+
+  describe "PUT player#update" do
+    it "update player" do
```

Comment:
minor typo
số ít thì tiếng Anh thường thêm s hay es kiểu như vì `it` là số ít
updates, deletes, returns, should.....